### PR TITLE
Setup RTC for Doubleclick

### PR DESCRIPTION
**Pull request by bradfrizzell**
see https://github.com/ampproject/amphtml/issues/8551

Still a WIP

Currently set up the basic callout that adds the targeting to the ad url, and a test page.

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -94,6 +98,12 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
   getAdUrl() {
     // TODO: Check for required and allowed parameters. Probably use
     // validateData, from 3p/3p/js, after noving it someplace common.
+    const rtcConfig = tryParseJson(
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: We should expect the config to be either meta tag or a JSON object defined in the head

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 2]
```diff
@@ -256,6 +292,39 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       this.attemptChangeSize(height, width).catch(() => {});
     }
   }
+
+  sendRtcRequestPromise(rtcConfig) {
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Make private

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 3]
```diff
@@ -256,6 +292,39 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+    try {
+      endpoint = rtcConfig['doubleclick']['endpoint'];
+    } catch (err) {
+      // report error
+      return;
+    }
+
+    let rtcResponse;
+
+    const xhr = new XMLHttpRequest();
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: use xhrFor(this.win).fetchJson

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 4]
```diff
@@ -256,6 +292,39 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+
+    xhr.open('GET', endpoint, true);
+    //xhr.withCredentials = true;
+    xhr.send();
+
+    const requestPromise = new Promise(resolve => {
+      rtcResponse = resolve;
+    });
+
+    const timeout = timerFor(window).timeoutPromise(10000);
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: 2 seconds is more reasonable and you will need to add error handling

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 2 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 3 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 4 above)
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

**Comment by bradfrizzell** [comment]
moved to https://github.com/ampproject/amphtml/pull/9857 due to changes from SRA that made rebasing to get changes too much of a pain

**Pull request closed without merging**